File: mobile/lib/models/student.dart
class StudentProfile {
  StudentProfile({
    required this.id,
    required this.studentNumber,
    required this.fullName,
    this.email,
    this.phone,
    this.address,
    this.guardianName,
    this.guardianPhone,
    this.className,
    this.department,
    this.session,
    this.faculty,
    this.level,
    this.photoUrl,
    this.bloodGroup,
    this.allergies,
    this.medicalNotes,
    required this.status,
    required this.walletBalance,
    required this.walletCurrency,
  });

  final int id;
  final String studentNumber;
  final String fullName;
  final String? email;
  final String? phone;
  final String? address;
  final String? guardianName;
  final String? guardianPhone;
  final String? className;
  final String? department;
  final String? session;
  final String? faculty;
  final String? level;
  final String? photoUrl;
  final String? bloodGroup;
  final String? allergies;
  final String? medicalNotes;
  final String status;
  final String walletBalance;
  final String walletCurrency;

  factory StudentProfile.fromJson(Map<String, dynamic> json) {
    return StudentProfile(
      id: json['id'] as int,
      studentNumber: json['student_number'] as String,
      fullName: json['full_name'] as String,
      email: json['email'] as String?,
      phone: json['phone'] as String?,
      address: json['address'] as String?,
      guardianName: json['guardian_name'] as String?,
      guardianPhone: json['guardian_phone'] as String?,
      className: json['class_name'] as String?,
      department: json['department'] as String?,
      session: json['session'] as String?,
      faculty: json['faculty'] as String?,
      level: json['level'] as String?,
      photoUrl: json['photo_url'] as String?,
      bloodGroup: json['blood_group'] as String?,
      allergies: json['allergies'] as String?,
      medicalNotes: json['medical_notes'] as String?,
      status: json['status'] as String,
      walletBalance: json['wallet_balance']?.toString() ?? '0.00',
      walletCurrency: json['wallet_currency'] as String? ?? 'NGN',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'student_number': studentNumber,
      'full_name': fullName,
      'email': email,
      'phone': phone,
      'address': address,
      'guardian_name': guardianName,
      'guardian_phone': guardianPhone,
      'class_name': className,
      'department': department,
      'session': session,
      'faculty': faculty,
      'level': level,
      'photo_url': photoUrl,
      'blood_group': bloodGroup,
      'allergies': allergies,
      'medical_notes': medicalNotes,
      'status': status,
      'wallet_balance': walletBalance,
      'wallet_currency': walletCurrency,
    };
  }
}

File: backend/app/Models/Student.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;
use Illuminate\Support\Facades\Storage;

class Student extends Model
{
    use HasFactory;

    protected $fillable = [
        'student_number',
        'first_name',
        'last_name',
        'email',
        'phone',
        'address',
        'guardian_name',
        'guardian_phone',
        'class_name',
        'department',
        'session',
        'faculty',
        'level',
        'photo_path',
        'date_of_birth',
        'gender',
        'blood_group',
        'allergies',
        'medical_notes',
        'status',
    ];

    protected function casts(): array
    {
        return [
            'date_of_birth' => 'date',
        ];
    }

    public function getFullNameAttribute(): string
    {
        return trim("{$this->first_name} {$this->last_name}");
    }

    public function getPhotoUrlAttribute(): ?string
    {
        return $this->photo_path ? Storage::disk('public')->url($this->photo_path) : null;
    }

    public function nfcCards(): HasMany
    {
        return $this->hasMany(NfcCard::class);
    }

    public function activeCard(): HasOne
    {
        return $this->hasOne(NfcCard::class)->where('status', 'active')->latestOfMany();
    }

    public function wallet(): HasOne
    {
        return $this->hasOne(Wallet::class);
    }

    public function clinicVisits(): HasMany
    {
        return $this->hasMany(ClinicVisit::class);
    }

    public function attendanceRecords(): HasMany
    {
        return $this->hasMany(AttendanceRecord::class);
    }

    public function examEligibilities(): HasMany
    {
        return $this->hasMany(ExamEligibility::class);
    }
}
